FeedStoreChallenge/CoreDataFeedStore/CoreDataFeedStore.swift

```diff
@@ -0,0 +1,122 @@
+// 
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+public class CoreDataFeedStore: FeedStore {
+	public typealias Model = CDFeedStoreModel
+	public typealias LoadedStoresCompletion = NSPersistentContainer.LoadedStoresCompletion
+	
+	private let container: NSPersistentContainer
+	private let context: NSManagedObjectContext
+	
+	public enum Error: Swift.Error {
+		case invalidModelURL
+		case invalidStoreURL
+		case modelNotFound(URL)
+		case loadPersistentStoresFailed([NSPersistentContainer.StoreLoadFailure])
+		case unhandled(Swift.Error?)
+	}
+
+	public init(
+		name: String = Model.name,
+		modelURL: URL? = Model.modelURL(),
+		storeURL: URL? = Model.storeURL(),
```

Comment:
The `storeURL` shouldn't be optional because this class cannot work with a `nil` URL.

```suggestion
		storeURL: URL = Model.storeURL(),
```